### ISSUE TITLE
Interface 'ButtonProps<T>' cannot simultaneously extend types 'AnchorHTMLAttributes<T>' and 'ButtonHTMLAttributes<T>'

### ISSUE THREAD

**Issue by dawsonc623** [material-components/material-components-web-react#781]
Due to a [change in `@types/react`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34194), the definition of `ButtonProps` does not work anymore. I have not dived into exactly what it would take to fix it, but the only workaround there is right now is for users to stick to `@types/react@16.8.8` or lower in their project's dependencies.

**Comment by ben-mckernan** [comment]
The easy fix would be to change it to use `HTMLProps` like it is done in icon-button.
https://github.com/material-components/material-components-web-react/blob/rc0.12.0/packages/icon-button/index.tsx#L45-L46

A more complex approach would be to have a conditional generic like what is done for input in text-field.
https://github.com/material-components/material-components-web-react/blob/rc0.12.0/packages/text-field/Input.tsx#L44-L47

**Comment by moog16** [comment]
@ben-mckernan yes you're spot on. It should be extending HTMLProps instead of HTMLAttributes. text-field has a good example of this.

I'm not sure what the exact issue is with extending both button and anchor, but conceptually you can't be both at the same time (so this doesn't make sense). If someone would like to open a PR updating the types feel free.

**Comment by Jason3S** [comment]
The issue is with the `type` property in both interfaces. They are not of the same type, so extends doesn't know how to merge them.

**Comment by moog16** [comment]
Right but this only works for the Button case...if you were using the anchor tag you would need to change this. This issue is to make sure developers can import ButtonProps and it works for both cases. :)
